specs/DefaultDownloadDialog.md

```diff
@@ -0,0 +1,272 @@
+  HRESULT OpenDefaultDownloadDialog();
+
+  /// Close the default download dialog.
+  HRESULT CloseDefaultDownloadDialog();
+
+  /// Get the default download dialog position.
+  [propget] HRESULT DefaultDownloadDialogPosition(
+      [out, retval] COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION* position);
+
+  /// Set the default download dialog position relative to the WebView bounds.
+  /// The dialog can be positioned against any of the WebView corners (see
+  /// COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION). When the WebView or dialog
+  /// changes size, the dialog keeps its position relative to the corner. The
+  /// dialog may become partially or completely outside of the WebView bounds if
+  /// the WebView is small enough. Set the padding from the position with the
+  ///  `DefaultDownloadDialogPadding` property.
+  ///
+  /// \snippet ViewComponent.cpp SetDefaultDownloadDialogPosition
+  [propput] HRESULT DefaultDownloadDialogPosition(
+      [in] COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION position);
+
+  /// Get the default download dialog padding.
+  [propget] HRESULT DefaultDownloadDialogPadding(
+      [out, retval] COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_PADDING* padding);
+
+  /// Set the default download dialog padding from the
+  /// `DefaultDownloadDialogPosition`. See
+  /// COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_PADDING.
+  [propput] HRESULT DefaultDownloadDialogPadding(
+      [in] COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_PADDING padding);
+}
+
+/// Implements the interface to receive `IsDefaultDownloadDialogOpenChanged`
+/// events.
+[uuid(86557854-794a-414a-b046-ba515f617306), object, pointer_default(unique)]
+interface ICoreWebView2IsDefaultDownloadDialogOpenChangedEventHandler :
+   IUnknown {
+  /// Provides the event args for the corresponding event. No event args exist
+  /// and the `args` parameter is set to `null`.
+  HRESULT Invoke([in] ICoreWebView2* sender, [in] IUnknown* args);
+}
+```
+
+```c# (but really MIDL3)
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2;
+
+    enum CoreWebView2DefaultDownloadDialogPosition
+    {
+        TopLeft = 0,
+        TopRight = 1,
+        BottomLeft = 2,
+        BottomRight = 3,
+    };
+
+    struct CoreWebView2DefaultDownloadDialogPadding
+    {
+        UInt32 x;
+        UInt32 y;
+    };
+
+    runtimeclass CoreWebView2
+    {
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2_6")]
+        {
+            // ICoreWebView2_6 members
+            Boolean IsDefaultDownloadDialogOpen { get; };
+
+            event Windows.Foundation.TypedEventHandler<CoreWebView2, Object>
+                IsDefaultDownloadDialogOpenChanged;
+
+            void OpenDefaultDownloadDialog();
+
+            void CloseDefaultDownloadDialog();
+
+            CoreWebView2DefaultDownloadDialogPosition
+                DefaultDownloadDialogPosition { get; set; };
+
+            CoreWebView2DefaultDownloadDialogPadding
```

Comment:
Do we need a new type here? Can we use POINT (in COM which should auto convert to Windows.Foundation.Point in WinRT and whatever the appropriate thing is in .NET)?

Reply:
(Or is this because I suggested using unsigned values? Apologies if so - we would probably rather reuse the existing point type even if its signed than make a new type to get unsigned)

Reply:
Yes I was trying to make it unsigned. Will change back :)